Docs/developer_guide/mrml_overview.md

```diff
@@ -100,14 +100,346 @@ The following methods on the MRML scene are used to manage Undo/Redo stacks:
+* Constructing a new node is same as the [data node](#the-data-node).
+* Override [`bool CanReadInReferenceNode(vtkMRMLNode *refNode)`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#af76f30cec1ccdc03bc496e5d39c784da) to allow a user to inquire at runtime if a particular node can be read in by this storage node.
+* Override protected [`void InitializeSupportedReadFileTypes()`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#af7c838cb215a8a5b1dd87a123b3fbc81) to show what file types and extensions this storage node can read (can be more than one).
+* Override protected [`void InitializeSupportedWriteFileTypes()`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#ab161ed9e9869cb61441659fe7c358906) to show what types and extensions this storage node can read (can be more than one).
+  * It is recommended to be able to read and write the same file types within a single storage node.
+* Override protected [`int ReadDataInternal(vtkMRMLNode *refNode)`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#a56aa52786dad2724a0a6a706b7fcf014):
+  * This is called by the public [`ReadData`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#afeeb4c95ee3164bd1d3374c053ffb04f) method.
+  * This is where the actually reading data from a file happens.
+* Override protected [`int WriteDataInternal(vtkMRMLNode *refNode)`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#a739831f001a7cebbeb72484944e842bf):
+  * This is called by the public [`WriteData`](http://apidocs.slicer.org/master/classvtkMRMLStorageNode.html#a1eb7a2b35d28175e2c490ac51cbf5baf) method.
+  * This is where the actually writing data to a file happens.
+* If your data node uses any coordinates (most nodes that get displayed should) it is recommended to be specific in your storage format whether the saved coordinates are RAS or LPS coordinates.
+  * Having a way to allow the user to specify this is even better.
+* Other methods
+  * Adding a `vtkMRML<MyCustomType>Node* Create<MyCustomType>Node(const char* nodeName)` function will be convenient for implementing the writer and is also convenient for users of the storage node.
+
+:::{tip}
+
+If your storage node reads/writes JSON, [RapidJSON](https://rapidjson.org/index.html) is already in the superbuild and is the recommended JSON parser.
+
+It is recommended to have your extension be `.<something>.json` where the `<something>` is related to your node type (e.g. `.mrk.json` for Markups).
+
+:::
+
+#### The reader
+
+The recommended way to read a file into a MRML node is through the storage node. The reader, on the other hand, exists to interface with the loading facilities of 3D Slicer (drag and drop, as well as the button to load data into the scene). As such, the reader uses the storage node in its implementation.
+
+
+Files:
+
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Reader.h
+* \<Extension>/\<Module>/qSlicer\<MyCustomType>Reader.cxx
```

Comment:
```
|-- <Extension>
       |-- <Module>
              |-- qSlicer<MyCustomType>Reader.h
              |-- qSlicer<MyCustomType>Reader.cxx
```

Reply:
Updated to suggested file structure formatting.